pkg/builder/service_builder.go

```diff
@@ -68,6 +68,9 @@ func (b *Builder) BuildService(mariadb *mariadbv1alpha1.MariaDB, key types.Names
 	if opts.SessionAffinity != nil {
 		svc.Spec.SessionAffinity = *opts.SessionAffinity
 	}
+	if opts.AllocateLoadBalancerNodePorts != nil {
+		svc.Spec.AllocateLoadBalancerNodePorts = *&opts.AllocateLoadBalancerNodePorts
```

Comment:
As the linter suggests, this can be simplified to:
```suggestion
		svc.Spec.AllocateLoadBalancerNodePorts = opts.AllocateLoadBalancerNodePorts
```
Is there any reason for not doing so?